packages/amplify-graphql-transformer-core/src/transformer-context/output.ts

```diff
@@ -0,0 +1,599 @@
+      if (node.kind === Kind.OBJECT_TYPE_DEFINITION) {
+        return node as ObjectTypeDefinitionNode;
+      }
+    }
+  }
+
+  /**
+   * Extends the context query object with additional fields.
+   * If the customer uses a name other than 'Query' this will proxy to the
+   * correct type.
+   * @param fields The fields to add the query type.
+   */
+  public addQueryFields(fields: FieldDefinitionNode[]) {
+    const queryTypeName = this.getQueryTypeName();
+    if (queryTypeName) {
+      if (!this.getType(queryTypeName)) {
+        this.addType(blankObject(queryTypeName));
+      }
+      let queryType = objectExtension(queryTypeName, fields);
+      this.addObjectExtension(queryType);
+    }
+  }
+
+  /**
+   * Extends the context mutation object with additional fields.
+   * If the customer uses a name other than 'Mutation' this will proxy to the
+   * correct type.
+   * @param fields The fields to add the mutation type.
+   */
+  public addMutationFields(fields: FieldDefinitionNode[]) {
+    const mutationTypeName = this.getMutationTypeName();
+    if (mutationTypeName) {
+      if (!this.getType(mutationTypeName)) {
+        this.addType(blankObject(mutationTypeName));
+      }
+      let mutationType = objectExtension(mutationTypeName, fields);
+      this.addObjectExtension(mutationType);
+    }
+  }
+
+  /**
+   * Extends the context subscription object with additional fields.
+   * If the customer uses a name other than 'Subscription' this will proxy to the
+   * correct type.
+   * @param fields The fields to add the subscription type.
+   */
+  public addSubscriptionFields(fields: FieldDefinitionNode[]) {
+    const subscriptionTypeName = this.getSubscriptionTypeName();
+    if (subscriptionTypeName) {
+      if (!this.getType(subscriptionTypeName)) {
+        this.addType(blankObject(subscriptionTypeName));
+      }
+      let subscriptionType = objectExtension(subscriptionTypeName, fields);
+      this.addObjectExtension(subscriptionType);
+    }
+  }
+
+  /**
+   * Add an object type extension definition node to the context. If a type with this
+   * name does not already exist, an exception is thrown.
+   * @param obj The object type definition node to add.
+   */
+  public addObjectExtension(obj: ObjectTypeExtensionNode) {
+    if (!this.nodeMap[obj.name.value]) {
+      throw new Error(`Cannot extend nonexistent type '${obj.name.value}'.`);
+    }
+    // AppSync does not yet understand type extensions so fold the types in.
+    const oldNode = this.getObject(obj.name.value);
+    assert(oldNode);
+    const newDirs = [];
+    const oldDirs = oldNode?.directives || [];
+
+    // Filter out duplicate directives, do not add them
+    if (obj.directives) {
+      for (const newDir of obj.directives) {
+        if (Boolean(oldDirs.find(d => d.name.value === newDir.name.value)) === false) {
+          newDirs.push(newDir);
+        }
+      }
+    }
+
+    const mergedDirs = [...oldDirs, ...newDirs];
```

Comment:
would rename to mergedDirectives, oldDirectives, newDirectives. Dirs seems like it should be short for "directories"